### PR TITLE
RFC: A real-time monitoring data capturer and player

### PR DESCRIPTION
**Pull request by nathanusask**
This RFC describes how we want to implement a system where we can capture data from surveillance cameras and play the video either as live streaming or as history.

--------------------
@jingzhongxu @luqz @Kayryu

### CODE REVIEW

File: rfc/RFC-ABM-210329.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+In the first phase of this project, we aim to build a system that captures data directly from surveillance cameras, plays the footages and stores the data on the ABMatrix's private IPFS network.
+
+
+## Proposal
+
+### Architecture design
+```
+      Network camera              Website Embedded     Viewer with
+        Source                    Video Player     IPFS Client
+          |                              ^             ^                   ^             ^
```
Comment (Kayryu) [inline rfc/RFC-ABM-210329.md]: User how to call 'ipfs-mirror'?

File: rfc/RFC-ABM-210329.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	- install `ffmpeg`
+	- install `go-ipfs` and start `IPFS daemon`
+
+- Server-side implementation layout
+	- FFMpeg controller: 
+		- connects the network camera and starts encoding to publishable format such as the `ts+m3u8` format combination that's suitable for HLS players embedded in mainstream browsers.
+	- File watcher:
+		- whenever there's a new stream file generated the watcher notifies the observers to update information accordingly
+	- IPFS controller:
+		- adds and pins data on the network
```
Comment (Kayryu) [inline rfc/RFC-ABM-210329.md]: I problem that why we need IPNS, if user want watch history video, he must to known which file to load, so he will query the file from server, then get file from ipfs

File: rfc/RFC-ABM-210329.md (hunk 1 above)
Comment (nathanusask) [reply rfc/RFC-ABM-210329.md]: via IPFS gateway

File: rfc/RFC-ABM-210329.md (hunk 2 above)
Comment (nathanusask) [reply rfc/RFC-ABM-210329.md]: IPNS publishes mutable content to the network with an immutable hash; users on the client-side can then resolve the immutable hash to get the newest content.

**Pull request merged**